examples/vision/siamese.py

```diff
@@ -0,0 +1,290 @@
+  ----------
+  dataset : TensorFlow Dataset object
+            The dataset to visualize, with batch of the form
+            ((image_1, image_2), label)
+
+  to_show : int
+             Number of examples to visualize (default is 6)
+           `to_show` must be an integral multiple of `num_col`.
+            If not, then it will be converted to that.
+
+  num_col : int
+            Number of images in one row - (default is 3)
+
+  predictions : list/array
+                Array of predictions with shape (to_show, 1) - (default None)
+                Must be passed when test=True
+
+  test  : boolean
+          Whether the dataset being visualized is train dataset or
+          test dataset - (default False)
+
+  """
+
+  # When a user inputs more columns than there are images to visualize:
+  num_row = to_show//num_col if to_show//num_col != 0 else 1 
+
+  # `to_show` must be an integral multiple of `num_col`  
+  to_show = to_show if to_show/num_col == 0 else num_row*num_col
+
+  # Plot the images
+  fig, axes = plt.subplots(num_row, num_col, figsize=(5,5))
+  for images, labels in dataset.take(1):
+    for i in range(to_show):
+
+        # If the number of rows is 1, the axes array is one-dimensional
+        if num_row == 1:
+          ax = axes[i%num_col]
+        else:
+          ax = axes[i//num_col, i%num_col]
+          
+        # images[0][i][:,:,0] -> because it is
+        # (28,28,1) and imshow takes (28,82)
+        ax.imshow(tf.concat([images[0][i],images[1][i]],axis=1), cmap='gray')
+        if test:
+          ax.set_title('True: {} | Pred: {:.5f}'.format(
+          labels[i],
+          predictions[i][0]))
+        else:
+          ax.set_title('Label: {}'.format(labels[i]))
+  if test:
+    plt.tight_layout(rect = (0,0,1.9,1.9 ), w_pad=0.0)
+  else:
+    plt.tight_layout(rect = (0,0,1.5,1.5))
+  plt.show()
+  
+visualize(train_ds)
+
+"""
+## Define the model
+
+There will be two input layers, each leading to its own network, which
+produces embeddings. Lambda layer will merge them using
+[Euclidean distance](https://en.wikipedia.org/wiki/Euclidean_distance) and the
+merged layer will be fed to final network.
+"""
+
+def euclidean_distance(vects):
+    x, y = vects
+    sum_square = K.sum(K.square(x - y), axis=1, keepdims=True)
+    return K.sqrt(K.maximum(sum_square, K.epsilon()))
+
+
+
+input = Input((28,28,1))
+x = tf.keras.layers.BatchNormalization()(input)
+x = Conv2D(4, (5,5), activation = 'tanh')(x)
+x = AveragePooling2D(pool_size = (2,2))(x)
+x = Conv2D(16, (5,5), activation = 'tanh')(x)
+x = AveragePooling2D(pool_size = (2,2))(x)
+x = Flatten()(x)
+
+x = tf.keras.layers.BatchNormalization()(x)
+x = Dense(10, activation = 'tanh')(x)
+embedding_network = Model(input, x)
+
+
+input_1 = Input((28,28,1))
+input_2 = Input((28,28,1))
+
+# As mentioned above Siamese Network share weights between
+# tower networks (sister networks). To allow this sharing we will use
+# same embedding network for both tower networks
```

Comment:
small suggestion wrt usage of commas and remove `sharing` word in second line ("To allow this ~sharing~"):
```python
# As mentioned above, Siamese Network share weights between
# tower networks (sister networks). To allow this, we will use
# same embedding network for both tower networks
```